bin/publish-to-registries.sh

```diff
@@ -18,28 +16,34 @@ bin/build.sh "${STACK_VERSION}"
     | docker login "$INTERNAL_REGISTRY_HOST" -u "$INTERNAL_REGISTRY_USERNAME" --password-stdin
 )
 
+date=$(date -u '+%Y-%m-%d-%H.%M.%S')
+datedTagSuffix=".${date}"
+publicRepo="heroku/heroku"
+privateRepo="heroku/heroku-private"
```

Comment:
I've opened https://github.com/heroku/base-images/pull/246 for this.

Reply:
With heroku-24, we have to build+push at the same time, so the temporary/dated tag was helpful. We could basically build+push to the private tags, and promote them to a stable public tag once both the run and build variant built successfully.

If I were to have heroku-24 build+push directly to `heroku/heroku:24` and `heroku/heroku:24-build`, there is a chance that the first one succeeds, but the second fails and we end up with a partial publish.

I think we'll still want something like this for the multi-arch images. Maybe `heroku/heroku:24_${date}`? Or maybe `heroku/heroku:24_${GITHUB_RUN_ID}`? I honestly think the private repo is pretty nice for this, since it prevents public access to potentially partial results.

Reply:
I'm using`heroku/heroku:${STACK_VERSION}.temp-${GITHUB_RUN_ID}` as the temp tag now. These get deleted at the end of the action now.